solutions/USACO dec20-abc-bronze.mdx

```diff
@@ -0,0 +1,108 @@
+---
+id: usaco-1059
+source: USACO Bronze 2020 December
+title: Do You Know Your ABCs?
+author: Mrinall Umasudhan
+---
+
+[Official Analysis (C++ and Python)](http://usaco.org/current/data/sol_prob1_bronze_dec20.html)
+
+
+### Implementation
+
+<LanguageSection>
+
+<JavaSection>
+
+```java
+ import java.io.*;
+ import java.util.*;
+
+public class abc{
+      public static int[] swap(int d[], int l, int r){
+           int temp = d[l];
+           d[l] = d[r];
+           d[r] = temp;
+           return d;
+      }
+      
+      public static int[] reverse(int data[], int left, int right){
+           while(left < right){
+                int temp = data[left];
+                data[left++] = data[right];
+                data[right--] = temp;
+           }
+           
+           return data;
+      }
+      
+      // Generate all possible permutations of the array.
+      
+      public static boolean findNextPermutation(int d[]){
+           if(d.length <= 1){
+                return false;
+           }
+         
+           int last = d.length - 2;
+
+           while(last >= 0){
+                if (d[last] < d[last + 1]) {
+                     break;
+                }
+                last--;
+           }
+           
+           if(last < 0){
+                return false;
+           }
+           
+           int nextGreater = d.length - 1;
+           for(int i = d.length - 1; i > last; i--){
+                if(d[i] > d[last]){
+                     nextGreater = i;
+                     break;
+                }
+           }
+          
+          d = swap(d, nextGreater, last);
+          d = reverse(d, last + 1, d.length - 1);
+          return true;
+    }
+
+    // Given the a permutation check if the values fufill the condition given in the problem statement.
+    public static boolean valid(int[] arr){ 
+         int a = arr[0]; int b = arr[1]; int c = arr[2];
+         int aPlusB = arr[3]; int bPlusC = arr[4]; int cPlusA = arr[5]; 
+         int totalVal = arr[6];
+         
+         
+         return (totalVal == (a + b + c) && 
+         aPlusB == (a + b) && 
+         bPlusC == (b + c) && 
+         cPlusA == (c + a))
```

Comment:
semicolon??!

Also, add another tab before each of these 3 lines (like hanging indent for multiline paragraphs)

Reply:
Got it.